### PR TITLE
Zoma/merging fix comments

### PR DESCRIPTION
**Pull request by coldplaying42**
Support inline object triggers

### CODE REVIEW

File: experimental/generation/generator/src/library/mergeAssets.ts [hunk 1]
```diff
@@ -222,12 +221,10 @@ async function mergeLUFiles(schemaName: string, oldPath: string, newPath: string
         }
     }
 
+    let index = oldText.indexOf(`${os.EOL}[`)
+    let startLines = oldText.substring(0, index)
```
Comment (chrimc62) [inline experimental/generation/generator/src/library/mergeAssets.ts:225]: This is not really solving the problem, right?  This is just including everything in the file until a [ and this is merging LU files--in theory people can put comments everywhere.

File: experimental/generation/generator/src/library/mergeAssets.ts [hunk 2]
```diff
@@ -703,6 +703,20 @@ async function mergeDialogs(schemaName: string, oldPath: string, newPath: string
...
 }
 
+/**
+ * @description: Get the trigger name
+ * @param trigger trigger from main.dialog file
+ */
+function getTriggerName(trigger: any): string {
+    let triggerName: string
+    if (typeof trigger !== 'string') {
+        triggerName = trigger['$id']
```
Comment (chrimc62) [inline experimental/generation/generator/src/library/mergeAssets.ts:713]: >$id [](start = 31, length = 3)

Please change this to 'id'.  I'm in the process of removing $id.

File: experimental/generation/generator/src/library/mergeAssets.ts (hunk 1 above)
Comment (coldplaying42) [reply experimental/generation/generator/src/library/mergeAssets.ts:225]: Fix the problem.

File: experimental/generation/generator/src/library/mergeAssets.ts (hunk 2 above)
Comment (coldplaying42) [reply experimental/generation/generator/src/library/mergeAssets.ts:713]: OK.

**Pull request merged**